frontend/src/metabase/styled-components/selectors.ts

```diff
@@ -20,7 +20,9 @@ export const getFont = createSelector(
 export const getFontFiles = createSelector(
   [getSettings, getEmbedOptions],
   (settings, embedOptions) => {
-    if (embedOptions.font) {
+    // If a font is set in the embed options, we don't need to load any font files.
+    // However, we still need to load the custom font files if the font is set to "Custom".
+    if (embedOptions.font && embedOptions.font !== "Custom") {
```

Comment:
With the way we're loading font files right now. If there's `settings["application-font-files"]` we should just return it. There's no reason to conditionally return this value since `application-font-files` only exists when Metabase font is set to `custom`. Also, it's okay to even leave font files in the CSS since if `font-family` isn't set to `Custom` these fonts shouldn't be loaded anyway.

Here's how we use `fontFiles` at the moment. https://github.com/metabase/metabase/blob/5638d797689f87861ee20a92dea326d988d2bc58/frontend/src/metabase/styled-components/containers/GlobalStyles/GlobalStyles.tsx#L73-L83

I suggest we remove this if block entirely and the selector would just simply returns `settings["application-font-files"]`

Also see a [related discussion.](https://metaboat.slack.com/archives/C505ZNNH4/p1715098187144839)

Reply:
@WiNloS Awesome - thanks Kelvin! I thought something was fishy when I read this conditions, but I was unsure about removing it altogether since I'm not sure how it would work in static embedding / interactive embedding.

Reply:
done!